crates/matrix-sdk-crypto/src/olm/group_sessions/mod.rs

```diff
@@ -128,35 +130,50 @@ impl TryFrom<ExportedRoomKey> for ForwardedRoomKeyContent {
     /// This will fail if the exported room key doesn't contain an Ed25519
     /// claimed sender key.
     fn try_from(room_key: ExportedRoomKey) -> Result<ForwardedRoomKeyContent, Self::Error> {
-        // The forwarded room key content only supports a single claimed sender
-        // key and it requires it to be a Ed25519 key. This here will be lossy
-        // conversion since we're dropping all other key types.
-        //
-        // This isn't yet a problem since no other key types exist, but still
-        // something that will need to be addressed sooner or later.
-        if let Some(SigningKey::Ed25519(claimed_ed25519_key)) =
-            room_key.sender_claimed_keys.get(&DeviceKeyAlgorithm::Ed25519)
-        {
-            if room_key.algorithm == EventEncryptionAlgorithm::MegolmV1AesSha2 {
-                Ok(ForwardedRoomKeyContent::MegolmV1AesSha2(
-                    ForwardedMegolmV1AesSha2Content {
+        match room_key.algorithm {
+            EventEncryptionAlgorithm::MegolmV1AesSha2 => {
+                // The forwarded room key content only supports a single claimed sender
+                // key and it requires it to be a Ed25519 key. This here will be lossy
+                // conversion since we're dropping all other key types.
+                //
+                // This isn't yet a problem since no other key types exist, but still
+                // something that will need to be addressed sooner or later.
+                if let Some(SigningKey::Ed25519(claimed_ed25519_key)) =
```

Comment:
With this branch / comment only being for megolm-v1, is this still relevant or is megolm-v1 feature-frozen now that we have v2?

Reply:
I would say Megolm v1 is feature frozen but yeah v2 fixes this, so the comment should be updated.